dotcom-rendering/src/web/components/CommercialMetrics.tsx

```diff
@@ -1,46 +1,45 @@
 import type { ABTest } from '@guardian/ab-core';
 import { tests } from '@frontend/web/experiments/ab-tests';
-import { useEffect, useState } from 'react';
 import { sendCommercialMetrics } from '@guardian/commercial-core';
+import { useOnce } from '@root/src/web/lib/useOnce';
 import { useAB } from '@guardian/ab-react';
 import { useDocumentVisibilityState } from '../lib/useDocumentHidden';
+import { useAdBlockInUse } from '../lib/useAdBlockInUse';
 
+// TODO disallow undefined browserIds by placing conditional in App.tsx
+// so that we wait to render this component until browserId is defined.
 export const CommercialMetrics: React.FC<{
 	pageViewId: string;
 	browserId: string | undefined;
 }> = ({ pageViewId, browserId }) => {
 	const ABTestAPI = useAB();
 	const visibilityState = useDocumentVisibilityState();
 
-	const [sentCommercialMetrics, setSentCommercialMetrics] = useState<boolean>(
-		false,
-	);
-
-	useEffect(() => {
-		if (visibilityState !== 'hidden') return;
-		if (sentCommercialMetrics) return;
+	const adBlockerInUse = useAdBlockInUse();
+	const isHidden = visibilityState === 'hidden';
 
+	useOnce(() => {
 		const testsToForceMetrics: ABTest[] = [];
-		const shouldForceMetrics = ABTestAPI.allRunnableTests(
-			tests,
-		).some((test) =>
-			testsToForceMetrics.map((t) => t.id).includes(test.id),
+		const shouldForceMetrics = ABTestAPI.allRunnableTests(tests).some(
+			(test) => testsToForceMetrics.map((t) => t.id).includes(test.id),
 		);
 		const userIsInSamplingGroup = Math.random() <= 1 / 100;
 		const isDev =
 			window.guardian.config.page.isDev ||
 			window.location.hostname.includes('localhost');
 
-		if (isDev || shouldForceMetrics || userIsInSamplingGroup)
-			setSentCommercialMetrics(
-				sendCommercialMetrics(pageViewId, browserId, Boolean(isDev)),
-			);
+		if (isDev || shouldForceMetrics || userIsInSamplingGroup) {
+			const args: [string, string | undefined, boolean, boolean?] =
+				adBlockerInUse === undefined
```

Comment:
The way `useOnce` works means `adBlockerInUse` can never be `undefined` here, so I think you can just pass it in as a simple parameter and we no longer need this check

Reply:
Good call - that's much simpler